prime-dotnet-webapi/Services/SoapService.cs

```diff
@@ -22,9 +21,40 @@ public class SoapServiceOperationTuner : IServiceOperationTuner
 
         public void Tune(HttpContext httpContext, object serviceInstance, SoapCore.ServiceModel.OperationDescription operation)
         {
+            // This must be the header used by proxies that first receive the client certificate
+            // (and pass on via HTTP Header)
+            const string CLIENT_CERT_HEADER = "X-SSL-CERT";
+
             if (serviceInstance is SoapService service)
             {
-                service.DocumentRoot = GetRequestBody(httpContext, Prefix, Uri, operation.Name);
+                string urlEncoded;
+                StringValues paramValue;
+                if (httpContext.Request.Headers.TryGetValue(CLIENT_CERT_HEADER, out paramValue))
+                {
+                    urlEncoded = paramValue[0];
+                }
+                else
+                {
+                    throw new ArgumentException("Client certificate expected to be provided.");
+                }
+
+                if (PrimeEnvironment.PlrIntegration.ClientCertThumbprint == null)
+                {
+                    throw new SystemException("Receiving system is not configured properly; please advise system administrator.");
+                }
+
+                string decoded = HttpUtility.UrlDecode(urlEncoded);
+                // https://stackoverflow.com/questions/65349878/c-sharp-convert-certificate-string-into-x509-certificate
```

Comment:
I feel the following lines are rather self explanatory and don't need this link. 
X509Certificate2 basically only has two types of data sources: certs stored on disk; and byte arrays.